torch/fx/experimental/proxy_tensor.py

```diff
@@ -75,7 +90,10 @@ def proxy_call(func_overload, args, kwargs=None):
         kwargs = {}
     func = func_overload.overloadpacket
     if func_overload in CURRENT_DECOMPOSITION_TABLE:
-        return CURRENT_DECOMPOSITION_TABLE[func_overload](*args, **kwargs)
+        global PROXY_DISPATCH_MODE
+        proxy_mode = enable_torch_dispatch_mode(PROXY_DISPATCH_MODE) if PROXY_DISPATCH_MODE else nullcontext()
```

Comment:
You don't need to do this. `proxy_call` is called from `__torch_dispatch__` on the ProxyMode object you're looking for. Pass it in as an argument.